thread_pthread.c

```diff
@@ -1788,4 +1789,36 @@ rb_nativethread_self(void)
     return pthread_self();
 }
 
+/* A function that wraps actual worker function, for pthread abstraction. */
+static void *
+mjit_worker(void *arg)
+{
+    void (*worker_func)(void) = arg;
+
+    if (pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL) != 0) {
+	fprintf(stderr, "Cannot enable cancelation in MJIT worker\n");
+    }
```

Comment:
It's nice to set thread name:
```c
#ifdef SET_CURRENT_THREAD_NAME
    SET_CURRENT_THREAD_NAME("ruby-mjitworker");
#endif
```
Note that the max size of the name is 16 byte including NUL byte of the string.
A natural name of the thread "ruby-mjit-worker" is 16 byte; it exceeds 1 byte.

Reply:
Thank you to catch, fixed https://github.com/ruby/ruby/pull/1782/commits/ddb2ff7303982b19adb0a23707722e3ca4ece1d2